src/language-css/parser-postcss.js

```diff
@@ -300,6 +300,33 @@ function parseNestedCSS(node, options) {
       node.raws = {};
     }
 
+    // `postcss@8` parse custom properties as `css-decl`
+    if (options.parser === "css" && node.type === "css-decl") {
+      const { prop, value } = node;
+      if (
+        prop.startsWith("--") &&
+        value.startsWith("{") &&
+        value.endsWith("}")
+      ) {
+        const textBefore = options.originalText.slice(
+          0,
+          node.source.start.offset
+        );
+        const nodeText =
+          "a".repeat(prop.length) +
+          options.originalText.slice(
+            node.source.start.offset + prop.length,
+            node.source.end.offset + 1
+          );
+        const fakeContent = textBefore.replace(/[^\n]/g, " ") + nodeText;
+        node.value = {
+          type: "css-rule",
+          nodes: parseCss(fakeContent, [], { ...options }).nodes[0].nodes,
+        };
+        return node;
+      }
+    }
```

Comment:
I think we should remove them in the future major release, it is invalid

Reply:
I was going to parse is as `value-unknown`, but seems it is a proposal, but abandoned

http://tabatkins.github.io/specs/css-apply-rule/#defining

Reply:
Yes, but it is violate CSS spec, so prefer to avoid this, because it can break some future CSS features